sphinxcontrib/datatemplates/directive.py

```diff
@@ -114,3 +61,63 @@ def run(self):
         node.document = self.state.document
         nested_parse_with_titles(self.state, result, node)
         return node.children
+
+
+class DataTemplateJSON(DataTemplateBase):
+    def _load_data(self, env, data_source):
+        rel_filename, filename = env.relfn2path(data_source)
```

Comment:
This line is being repeated in every _load_data() method. Should the caller do that work? That would change the API of the method to only require the full filename.

Reply:
changed